pages/tapping/index.js

```diff
@@ -0,0 +1,348 @@
+  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.2);
+  &:hover {
+    transform: scale(1.05);
+  }
+`;
+
+const SpanContainer = styled.section`
+  background-image: url("/images/silver.avif");
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  text-align: center;
+  margin-top: 10px;
+
+  gap: 10px;
+  border-radius: 15px;
+  padding: 5px;
+  width: 310px;
+  height: 45px;
+`;
+
+const ButtonsContainer = styled.section`
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 15px;
+  width: 310px;
+  height: 45px;
+  margin-top: 5px;
+`;
+
+const StyledTappingButtonLink = styled(Link)`
+  background-image: url("/images/backbutton.png");
+  background-color: transparent;
+  background-size: contain;
+  background-position: center;
+  background-repeat: no-repeat;
+  color: transparent;
+  width: 70px;
+  height: 50px;
+
+  border: hidden;
+
+  cursor: pointer;
+  &:hover {
+    transform: scale(1.2);
+  }
+`;
+
+const StyledStartButton = styled.button`
+  background-image: url("/images/playbutton.png");
+  background-color: transparent;
+  background-size: contain;
+  background-position: center;
+  background-repeat: no-repeat;
+  color: transparent;
+  width: 70px;
+  height: 50px;
+  font-size: small;
+  border: hidden;
+
+  cursor: pointer;
+  &:hover {
+    transform: scale(1.2);
+  }
+`;
+
+const StyledPauseButton = styled(StyledStartButton)`
+  background-image: url("/images/pausebutton.png");
+`;
+
+const StyledResetButton = styled(StyledStartButton)`
+  background-image: url("/images/resetbutton.png");
+`;
+
+const SpeedUpMessage = styled.span`
+  position: absolute;
+  top: 200px;
+  left: 50%;
+  transform: translateX(-50%);
+  background-image: url("/images/orange.jpg");
+  background-size: cover;
+  background-repeat: no-repeat;
+  color: #fff;
+  font-size: 1.5rem;
+  padding: 10px 20px;
+  border-radius: 10px;
+  animation: fadeInOut 2s ease;
+  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.2);
+  z-index: 1000;
+
+  @keyframes fadeInOut {
+    0% {
+      opacity: 0;
+    }
+    20% {
+      opacity: 1;
+    }
+    80% {
+      opacity: 1;
+    }
+    100% {
+      opacity: 0;
+    }
+  }
+`;
+
+const CountdownMessage = styled(SpeedUpMessage)`
+  top: 275px;
+  z-index: 1000;
+  background-image: url("/images/red.jpg");
+  font-size: 4rem;
+`;
+
+export default function TappingGame({
+  petCollection,
+  currentPet,
+  onUpdatePetIndicator,
+}) {
+  const [activeCircles, setActiveCircles] = useState([]);
+  const [activeWrongCircles, setActiveWrongCircles] = useState([]);
+  const [score, setScore] = useState(0);
+  const [gameStarted, setGameStarted] = useState(false);
+  const [intervalTime, setIntervalTime] = useState(1600);
+  const [clickAllowed, setClickAllowed] = useState(true);
+  const [speedUpMessage, setSpeedUpMessage] = useState(false);
+  const [highScore, setHighScore] = useState(0);
+  const [countdown, setCountdown] = useState(60);
+  const [timeUpMessage, setTimeUpMessage] = useState(false);
```

Comment:
I suggest rethinking the overall game logic. You are using quite a lot of states here, many of which I believe are not all necessary. This makes the code unnecessarily complex and harder to read. A good practice is to use only as many states as are really needed. If we can derive a value from an already existing state, we don’t need a separate state for it. A good example would be the `speedUpMessage` and `timeUpMessage` states. Both of these can likely be derived from the `countdown` (i.e., when `countdown === 0`, the “time up” message is displayed).